content/blog/2022-12-08-announcing-poetry-1-3-0.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: single
+title: "Announcing Poetry 1.3.0"
+date: 2022-12-08
+categories: [releases]
+tags: ["1.x", "1.3"]
+---
+
+The Poetry team is pleased to announce the immediate availability of Poetry **1.3.0**.
+
+<!--more-->
+
+If you have a previous version of Poetry installed via the [official installer](/docs/#installation),
+getting Poetry **1.3.0** is as easy as:
+
+```bash
+$ poetry self update
+```
+
+## Highlights
+
+### Lock file 2.0
+
+Poetry 1.3.0 comes with lockfile version 2.0. It allows to map the locked distributions of
+multiple-constraints dependencies to the correct dependency especially if the dependencies
+have the same version and only differ in their type or source.
+
+While the change might seem breaking, Poetry 1.2.2 was previously prepared for this and can read
+new lockfile format without a problem.
+
+### No more corrupted cache
+
+With Poetry 1.3 we have fixed an issue, where package installations hung forever due to corrupted cache.
+
+### Scheduled change of `generate-setup-file`
+
+For a long time, when building a package, Poetry was generating `setup.py` file for compatibility with legacy
+`pip` installation method. With Poetry 1.4 and onwards, this option will be now off by default. If you wish to
+continue generating `setup.py` file on build, add following setting to your `pyproject.toml`:
+
+```toml
+[tool.poetry.build]
+generate-setup-file = true
+```
```

Comment:
```suggestion
## Highlights

### New lock file format

The lock file format has been changed to better support explicit sources. This change should be mostly transparent for
users, as Poetry 1.2.2 or newer are capable of reading the new format, and Poetry 1.3.0 maintains full support for
reading previous versions of the lock file format.

The lock file format had to be changed as previously filenames and hashes were locked in a global table. This could
cause collisions with explicit sources or other more advanced uses of Poetry's features, and the new format mitigates
this by instead locking files with the corresponding metadata table.

### Preventing cache corruption

Poetry 1.1 and 1.2 are prone to corrupting various caches in the following common scenarios:

* Interrupting an instance of Poetry while it is gathering metadata from PyPI.
* Concurrent instances of Poetry writing to the artifacts cache (e.g. `poetry install`, `poetry update`, etc).

The former typically manifested as an indefinite hang while making HTTP requests, and should be prevented entirely on
Poetry 1.3, regardless of the current state of your cache. This is because the fix has involved moving to a new
file-locking implementation, and stale locks from the old implementation will be ignored. However, if you wish to
explicitly clear this cache, you can do so with `poetry cache clear --all .`.

The latter improvement will not fix already-corrupted archives on disk; if you are experiencing hash mismatches, make
sure to try clearing the artifacts cache with `rm -rf $(poetry config cache-dir)`, and file a new issue if this still
occurs on Poetry 1.3.

## Upcoming Changes

### `generate-setup-file = false`

Poetry has long generated a stub `setup.py` file for compatibility with older versions of `pip` (specifically, `pip`
versions older than 19, the [first to introduce support for PEP 517][pip-19]). Users who did not want Poetry to do this
have long been able to opt out with the `generate-setup-file` option in `tool.poetry.build`.

The next minor release of Poetry will invert this behavior, turning off generation of `setup.py` by default. This change
should be transparent to nearly all users, as versions of `pip` supporting alternate build systems natively have been
available for nearly four years.

Users who wish to maintain the legacy behavior may explicitly do so in their `pyproject.toml`; however, users should
keep in mind that this option will be deprecated and eventually removed in future versions of Poetry:

```toml[tool.poetry.build] generate-setup-file = true ```

  [pip-19]: https://discuss.python.org/t/pip-19-0-is-now-available
```